src/Components/CreateImageWizard/ValidatedTextInput.tsx

```diff
@@ -51,16 +51,40 @@ interface HookValidatedTextInputWithButtonPropTypes
   isEmpty: boolean;
 }
 
-const getValidatedState = (
-  isPristine: boolean,
-  isEmpty: boolean | undefined,
-  stepValidation: StepValidation,
-  fieldName: string
-): 'default' | 'error' | 'success' => {
-  if (isEmpty) return 'default';
-  if (isPristine) return 'default';
-  if (stepValidation.errors[fieldName] === 'default') return 'default';
-  return stepValidation.errors[fieldName] ? 'error' : 'success';
+const useValidatedInput = (value: string, isEmpty?: boolean) => {
+  const [isPristine, setIsPristine] = useState(!value);
+
+  const handleBlur = () => {
+    setIsPristine(false);
+  };
+
+  const getValidationState = (
+    stepValidation: StepValidation,
+    fieldName: string
+  ) => {
+    if (isEmpty) return 'default';
+    if (isPristine) return 'default';
+    if (stepValidation.errors[fieldName] === 'default') return 'default';
+    return stepValidation.errors[fieldName] ? 'error' : 'success';
+  };
+
+  return {
+    handleBlur,
+    getValidationState,
+  };
+};
+
+const renderValidationHelperText = (
+  validationState: 'error' | 'default' | 'success',
```

Comment:
There are three possible validation states but only one of them ('error') is actually used. This seems like a good time to apply the "You aren't going to need it" (YAGNI) principle and simplify the component. Why not make this something like ErrorHelperText? Then the only prop needed would be errorMessage.